src/main/java/com/avispl/symphony/dal/communicator/aver/ptz/AverPTZCommunicator.java

```diff
@@ -0,0 +1,1885 @@
+				}
+				break;
+			}
+			default: {
+				throw new IllegalStateException("Unexpected value: " + exposureCommand);
+			}
+		}
+	}
+
+	/**
+	 * This method is used to control power
+	 */
+	public void powerControl(PowerStatus powerStatus) {
+		try {
+			int currentSeqNum = ++sequenceNumber;
+			byte[] response = send(buildSendPacket(cameraID, currentSeqNum, PayloadType.COMMAND.getCode(), CommandType.COMMAND.getCode(), PayloadCategory.CAMERA.getCode(),
+					Command.POWER.getCode(), powerStatus.getCode()));
+
+			digestResponse(response, currentSeqNum, CommandType.COMMAND, null);
+		} catch (Exception e) {
+			if (this.logger.isDebugEnabled()) {
+				this.logger.debug("error during power " + powerStatus.getName() + " send", e);
+			}
+		}
+	}
+
+	/**
+	 * This method is used to control zoom
+	 */
+	public void zoomControl(ZoomControl zoomControl) {
+		try {
+			int currentSeqNum = ++sequenceNumber;
+			byte[] response = send(buildSendPacket(cameraID, currentSeqNum, PayloadType.COMMAND.getCode(), CommandType.COMMAND.getCode(), PayloadCategory.CAMERA.getCode(),
+					Command.ZOOM.getCode(), (byte) (zoomControl.getCode() + zoomSpeed)));
+
+			digestResponse(response, currentSeqNum, CommandType.COMMAND, null);
+		} catch (Exception e) {
+			if (this.logger.isDebugEnabled()) {
+				this.logger.debug("error during zoom " + zoomControl.getName() + " send", e);
+			}
+		}
+
+		zoomStop();
+	}
+
+	/**
+	 * This method is used to control zoom stop
+	 */
+	public void zoomStop() {
+		try {
+			int currentSeqNum = ++sequenceNumber;
+			byte[] response = send(buildSendPacket(cameraID, currentSeqNum, PayloadType.COMMAND.getCode(), CommandType.COMMAND.getCode(), PayloadCategory.CAMERA.getCode(),
+					Command.ZOOM.getCode(), ZoomControl.STOP.getCode()));
+
+			digestResponse(response, currentSeqNum, CommandType.COMMAND, null);
+		} catch (Exception e) {
+			if (this.logger.isDebugEnabled()) {
+				this.logger.debug("error during zoom stop send", e);
+			}
+		}
+	}
+
+	/**
+	 * This method is used to control focus
+	 */
+	public void focusControl(FocusControl focusControl) {
+		try {
+			int currentSeqNum = ++sequenceNumber;
+			byte[] response = send(buildSendPacket(cameraID, currentSeqNum, PayloadType.COMMAND.getCode(), CommandType.COMMAND.getCode(), PayloadCategory.CAMERA.getCode(),
+					Command.FOCUS.getCode(), (byte) (focusControl.getCode() + focusSpeed)));
+
+			digestResponse(response, currentSeqNum, CommandType.COMMAND, null);
+		} catch (Exception e) {
+			if (this.logger.isDebugEnabled()) {
+				this.logger.debug("error during focus " + focusControl.getName() + " send", e);
+			}
+		}
+
+		focusStop();
+	}
+
+	/**
+	 * This method is used to control focus stop
+	 */
+	public void focusStop() {
+		try {
+			int currentSeqNum = ++sequenceNumber;
+			byte[] response = send(buildSendPacket(cameraID, currentSeqNum, PayloadType.COMMAND.getCode(), CommandType.COMMAND.getCode(), PayloadCategory.CAMERA.getCode(),
```

Comment:
similar code segment seems to be repeated multiple times. Is there a way to make it more abstract to reduce the amount of copy-pasted lines?

Reply:
ok I will fix it